### PR TITLE
api account creation and message reporting

### PR DESCRIPTION
**Pull request by williamhorning**
this adds the following endpoints:
- /auth/register (gen_account in cl)
- /auth/login (authpswd)
- /posts/post-id/report (report)
- /users/username/report (report)

### CODE REVIEW

**Comment by NotFenixio** [comment]
This is just awesome!

File: rest_api/auth.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    
+    if registration_blocked_ips.search_best(request.ip):
+        security.ratelimit(f"register:{request.ip}:f", 5, 30)
+        return {"error": True, "type": "registrationBlocked"}, 403
+
+    if security.account_exists(data.username):
+        security.ratelimit(f"register:{request.ip}:f", 5, 30)
+        return {"error": True, "type": "usernameExists"}, 409
+    
+    print("test")
```
Comment (tnix100) [inline rest_api/auth.py:84]: debug line, please remove

File: rest_api/auth.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    if (data.password not in account["tokens"]) and (not security.check_password_hash(data.password, account["pswd"])):
+        security.ratelimit(f"login:u:{data.username}:f", 5, 60)
+        abort(401)
+
+    db.netlog.update_one({"_id": {"ip": request.ip, "user": data.username}}, {"$set": {"last_used": int(time())}}, upsert=True)
+
+    security.ratelimit(f"login:u:{data.username}:s", 25, 300)
+
+    token = security.generate_token()
+
```
Comment (tnix100) [inline rest_api/auth.py:55]: there used to be something around here that sent an inbox message if an account deletion was cancelled, that should be re-implemented here

File: rest_api/users.py [hunk 3]
```diff
@@ -133,10 +140,51 @@ async def update_relationship(username, data: UpdateRelationshipBody):
         }
     }, direct_wrap=True, usernames=[request.user])
 
-    # Return updated relationship
+    # Return updated relationshipFeat/api accounts
```
Comment (tnix100) [inline rest_api/users.py:143]: I don't think this comment change was intentional

**Review by tnix100 — CHANGES_REQUESTED** [review]

File: rest_api/auth.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        "$addToSet": {"tokens": token},
+        "$set": {"last_seen": int(time()), "delete_after": None}
+    })
+
+    return {"error": False, "token": token, "account": security.get_account(data.username, True)}, 200
+
+@auth_bp.post('/register')
+@validate_request(AuthRequest)
+async def register(data: AuthRequest):
+    print("test")
```
Comment (tnix100) [inline rest_api/auth.py:66]: another debug line :p

**Review by tnix100 — APPROVED** [review]

**Pull request merged**